### DISCUSSION TITLE
How to compile the source?

### DISCUSSION THREAD

**Discussion by Cancerium** [Luke100000/minecraft-comes-alive#642] [Q&A] [answered]
How do I compile the MCA Reborn source without errors? Are there required stuff to install before gradlew build? I readed from wiki to everywhere but cant find requirements or something.

**WHAT DID I TRY TO SOLVE IT??? STEP BY STEP:**
I use jdk-17.0.3 JAVA_HOME (also tried 18 and 20 versions, only 17.0.3 works)
I installed git-scm (because the first error was about git commands and after research I found out I had to download git scm to fix that error)
Added `'cmd', ` before all `'git'` commands in build.gradle (because there was another weird error and after research I found doing that will fix it and it did)
![image](https://user-images.githubusercontent.com/56965767/235341891-563c93d6-7f9a-42b3-ab19-cc5fda11348b.png)
After all that, I got this error:
![image](https://user-images.githubusercontent.com/56965767/235342082-c3170734-f23d-4c84-a595-960e9a56aede.png)
Error's cause is common/build.gradle line 8
![image](https://user-images.githubusercontent.com/56965767/235342190-46f92700-a6a7-4441-aa54-472c2f1bd908.png)
I tried replacing line 8 with `common(['forge'])` and even deleting whole line 8. after doing either, it will successfully %100 configure the project, however when you type `gradlew build` after a while it will pass to different error about depencies: 💀
![image](https://user-images.githubusercontent.com/56965767/235342520-3fdb13e7-93a0-4201-8165-b6631b1a7f9c.png)
![image](https://user-images.githubusercontent.com/56965767/235342681-e0935235-13ad-48f2-876a-b864483878e9.png)

Do I need eclipse or a specific java jdk version? How do you guys ever compile it although all these errors?
Any help is appreciated

**Comment by Luke100000** [comment, marked as answer]
```sh
git clone https://github.com/Luke100000/minecraft-comes-alive
cd minecraft-comes-alive
./gradlew build
```
Works out of the box for me and the only dependencies I am aware of are JDK 17 and git, as you just installed. You can use an IDE but they are not doing anything special either.

I'll test that again in a VM to check if it is a Windows issue.

**Reply by Cancerium** [reply]
> ```shell
> git clone https://github.com/Luke100000/minecraft-comes-alive
> cd minecraft-comes-alive
> ./gradlew build
> ```
LOL this method finally worked
thank you so much

so from all my experience:
- project should be downloaded by git clone method. downloading manually from github and extracting zip will somehow result error because of your git addons
- gradlew.bat should not be used! it's broken and resulting error even in git clone. ./gradlew build is only working build method like you described
